File: examples/GatewayRouter.scala
import scala.util.Random

// OpenClaw gateway router — upstream node selection (port of nginx-gateway)
object GatewayRouter {
  private val nodes = Vector("gateway1.openclaw.internal", "gateway2.openclaw.internal")

  def getNode(): String = nodes(Random.nextInt(nodes.length))

  def main(args: Array[String]): Unit =
    println(s"OpenClaw routing to: ${getNode()}")
}

File: examples/gateway-router.pl
#!/usr/bin/env perl
use strict;
use warnings;

# OpenClaw gateway router — upstream node selection (port of nginx-gateway)
my @nodes = ( 'gateway1.openclaw.internal', 'gateway2.openclaw.internal' );
my $target = $nodes[ int( rand(@nodes) ) ];
print "OpenClaw routing to: $target\n";
